src/octree.jl

```diff
@@ -0,0 +1,154 @@
+
+struct OctreeQuantization <: AbstractColorQuantizer
+    numcolors::Int
+    function OctreeQuantization(
+        colorspace::Type{<:Colorant},
+        numcolors::Int = 256;
+        kwargs...,
+    )
+        colorspace == RGB{N0f8} || error("Octree Algorithm only supports RGB colorspace")
+        return new(numcolors)
+    end
+end
+
+const OCTREE_DEFAULT_COLORSPACE = RGB{N0f8}
+# Color to RGB
+# constructor for struct
+function OctreeQuantization(numcolors::Int = 256; kwargs...)
+    return OctreeQuantization(OCTREE_DEFAULT_COLORSPACE, numcolors; kwargs...)
+end
+
+function (alg::OctreeQuantization)(img::AbstractArray)
+    return octreequantisation!(img; numcolors = alg.numcolors)
+end
+
+function octreequantisation!(img; numcolors = 256, precheck::Bool = false)
+    # ensure the img is in RGB colorspace
+    if (eltype(img) != RGB{N0f8})
+        error("Octree Algorithm requires img to be in RGB colorspace")
+    end
+
+    # checks if image has more colors than in numcolors
+    if precheck == true
+        unumcolors = length(unique(img))
+        # @show unumcolors
+        if unumcolors <= numcolors
+            @debug "Image has $unumcolors unique colors"
+            return unique(img)
+        end
+    end
+
+    # step 1: creating the octree
+    root = Cell(
+        SVector(0.0, 0.0, 0.0),
+        SVector(1.0, 1.0, 1.0),
+        ["root", 0, [], RGB{N0f8}.(0.0, 0.0, 0.0), 0],
+    )
+    cases = map(
+        p -> [bitstring(UInt8(p))[6:end], 0, Vector{Int}([]), RGB{N0f8}.(0.0, 0.0, 0.0), 1],
+        1:8,
+    )
+    split!(root, cases)
+    inds = collect(1:length(img))
+
+    function putin(root, in)
+        r, g, b = map(p -> bitstring(UInt8(p * 255)), channelview([img[in]]))
+        rgb = r[1] * g[1] * b[1]
```

Comment:
I don't use it as a way to identify color but as a identifier of case number of which possible cases are ["000","001"...."111"] eight different cases of octree at each level. More details on this can be found here: http://delimitry.blogspot.com/2016/02/octree-color-quantizer-in-python.html